File: app/src/main/java/im/point/dotty/db/UserDao.java
package im.point.dotty.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import im.point.dotty.model.User;
import io.reactivex.Flowable;

@Dao
public interface UserDao {
    @Insert
    void addUser(User user);

    @Update
    void updateUser(User user);

    @Query("SELECT * FROM users WHERE id = :id")
    Flowable<User> getUser(long id);
}

File: app/src/main/java/im/point/dotty/db/UserDao.kt
/*
 * Copyright (c) 2019-2021 Pavel A. Sokolov
 */
package im.point.dotty.db

import androidx.room.Dao
import androidx.room.Query
import im.point.dotty.model.User
import kotlinx.coroutines.flow.Flow

@Dao
interface UserDao : CommonDao<User> {
    @Query("SELECT * FROM users WHERE id = :id")
    fun getItem(id: Long): User?

    @Query("SELECT * FROM users")
    fun getAllFlow(): Flow<List<User>>

    @Query("SELECT * FROM users WHERE id = :id")
    fun getItemFlow(id: Long): Flow<User?>
}